src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -880,5 +886,37 @@ public async Task Ping_TimedOut_TAP_Success()
             PingReply reply = await sender.SendPingAsync(TestSettings.UnreachableAddress);
             Assert.Equal(IPStatus.TimedOut, reply.Status);
         }
+
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [Theory]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        [InlineData(AddressFamily.InterNetwork)]
+        [InlineData(AddressFamily.InterNetworkV6)]
+        [OuterLoop] // Depends on sudo
+        public void SendPingWithIPAddressAndTimeoutAndBufferAndPingOptions_ElevatedUnix(AddressFamily addressFamily)
+        {
+            IPAddress localIpAddress = TestSettings.GetLocalIPAddress(addressFamily);
+            if (localIpAddress == null)
+            {
+                // No local address for given address family.
+                return;
+            }
+
+            _output.WriteLine($"pinging '{localIpAddress}'");
+
+            using (RemoteExecutor.Invoke(address  =>
+            {
+                byte[] buffer = TestSettings.PayloadAsBytes;
+                SendBatchPing(
+                    (ping) => ping.Send(address, TestSettings.PingTimeout, buffer, new PingOptions()),
+                    (pingReply) =>
+                    {
+                        PingResultValidator(pingReply, new IPAddress[] { IPAddress.Parse(address) }, null);
+                        Assert.Equal(buffer, pingReply.Buffer);
+                    });
+            }, localIpAddress.ToString(), new RemoteInvokeOptions { RunAsSudo = true }))
```

Comment:
You can just use `.Dispose()` rather than the empty using block.

Reply:
I like that better. I copy test structure from other assembly.